clash-cores/src/Clash/Cores/Xilinx/BlockRam.hs

```diff
@@ -0,0 +1,75 @@
+{-|
+  Copyright   :  (C) 2023 QBayLogic B.V.
+  License     :  BSD2 (see the file LICENSE)
+  Maintainer  :  QBayLogic B.V. <devops@qbaylogic.com>
+
+  Xilinx Block RAM primitives.
+-}
+
+-- See [Note: eta port names for tdpbram]
+{-# OPTIONS_GHC -fno-do-lambda-eta-expansion #-}
+
+module Clash.Cores.Xilinx.BlockRam (tdpbram) where
+
+import Clash.Explicit.Prelude
+import Clash.Cores.Xilinx.BlockRam.Internal (tdpbram#)
+
+import GHC.Stack (HasCallStack)
+
+-- | Instantiates a Block Memory Generator IP as described in PG058.
+--
+-- Any value that is being written on a particular port is also the
+-- value that will be read on that port, i.e. the same-port read/write behavior
+-- is: WriteFirst. For mixed-port read/write, when port A writes to the address
+-- port B reads from, the output of port B is undefined, and vice versa.
+tdpbram ::
+  forall nAddrs domB domA nBytes a .
```

Comment:
Why is `domB` before `domA`?

Reply:
Very good question. Changing it would be an API change (at least for the public ones). I'll change it for the internal ones in any case.